### ISSUE TITLE
Bach: support arbitrary names for Series

### ISSUE THREAD

**Issue by thijs-obj** [objectiv/objectiv-analytics#1209]
Names for Series are currently limited by the column names that a database supports, because Series names map directly to column names.

We want to remove this limitation.

Discussed approach:
- `Series` is responsible for going from 'real-name' to 'column-name'
- Column name translation algorithm should work both ways
- Should be transparent to user (as long as user doesn't look at or use the generated SQL)

**Comment by thijs-obj** [comment]
Make sure to enable the test `bach/tests/functional/bach/test_df_setitem.py::test_set_series_column_name_with_spaces()` for all databases afterwards

**Comment by ivarpruijn** [comment]
Is this issue resolved @thijs-obj ?

**Comment by thijs-obj** [comment]
Yes. There is still one sub-issue open https://github.com/objectiv/objectiv-analytics/issues/1317, but it's low prio and the main issue can be considered to be solved.
